### PR TITLE
[WFLY-9087] SharedSessionFailoverTestCase URLs resolving fixed

### PR DESCRIPTION
**Pull request by rhatlapa**
URLs were incorrectly resolved all pointing to war2 deployment instead
of one group to war1 and second group to war2 deployment.

Fixes https://issues.jboss.org/browse/WFLY-9087

### CODE REVIEW

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/shared/SharedSessionFailoverTestCase.java [hunk 1]
```diff
@@ -92,14 +91,16 @@
 
     @Test
     public void test(
-            @ArquillianResource(SimpleServlet.class) @OperateOnDeployment(DEPLOYMENT_1) URL baseURL1,
-            @ArquillianResource(SimpleServlet.class) @OperateOnDeployment(DEPLOYMENT_2) URL baseURL2)
+            @ArquillianResource @OperateOnDeployment(DEPLOYMENT_1) URL baseURLDep1,
+            @ArquillianResource @OperateOnDeployment(DEPLOYMENT_2) URL baseURLDep2)
             throws URISyntaxException, IOException {
+        URL baseURL1 = new URL(baseURLDep1.toExternalForm() + "/");
```
Comment (pferraro) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/shared/SharedSessionFailoverTestCase.java]: I'd prefer that we add SimpleServlet.createURI(...) methods that accept a URI which should avoid the need for the convoluted URL -> string -> URL -> string -> URL conversions.

**Review by pferraro — CHANGES_REQUESTED** [review]

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/shared/SharedSessionFailoverTestCase.java (hunk 1 above)
Comment (rhatlapa) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/shared/SharedSessionFailoverTestCase.java]: @pferraro thank you for review, I have reworked the changes to use more URI based approach and to limit URL -> String -> URL conversions. Done in separate commit. If it looks good to you, I shall squash the commits.
